### DISCUSSION TITLE
Migrate existing development site from XAMPP to Laragon

### DISCUSSION THREAD

**Discussion by bencote2020** [leokhoa/laragon#870] [Q&A]
I'm currently developing a website on XAMPP, but I'm interested in switching to Laragon. Can you provide step-by-step instructions on how to migrate my existing development site from XAMPP to Laragon? Additionally, please highlight any potential issues that may arise during the migration process and offer suggestions on how to resolve them.

**Comment by AWqxKAWERbXo** [comment]
Sorry, that's a bit too much of a favor imho. Also, it depends how you configured XAMPP. Also, your software (CMS, etc.) needs to be reconfigured if you made use of custom hosts.

The steps I'd need for that would be: 

1. backup everything in xampp
2. esp. make a full export of your database with your favorite tool like mysqldump, HeidiSQL (GUI). Follow their instructions to export every database with every table and data.
3. install laragon
4. if needed (depending on your previous XAMPP installation), update components like Apache, MySQL and PHP <https://github.com/leokhoa/laragon/wiki>
5. configure laragon settings, e.g. SSL, etc.
6. copy all www-files of xampp to the new location (usually C:\laragon\www\)
7. stop xampp
8. start laragon
9. import your database export using mysqldump or HeidiSQL
10. happily enjoy using laragon

**Reply by bencote2020** [reply]
Thank you for your reply and detailed instructions. XAMPP keeps on blowing up mySQL and I must keep replacing the data folder from backup to make it start again. I trust Laragon will provide more stability to mySQL..

**Reply by AWqxKAWERbXo** [reply]
That's the same reason i left xampp, so yes, I guess you will gain more stability.

**Comment by luisAntonioLAGS** [comment]
In this link you will find the steps to migrate from XAMPP
https://laragon.org/download/migrate-from-xampp
